wallet/src/de/schildbach/wallet/ui/main/WalletActivity.java

```diff
@@ -830,17 +836,17 @@ public void onNewKeyChainEncrypted() {
     }
 
     private void updateSyncState() {
-        if (blockchainState == null) {
-            return;
-        }
+        Boolean isSyncFailed = viewModel.isBlockchainSyncFailed().getValue();
 
-        if (blockchainState != null && blockchainState.syncFailed()) {
+        if (isSyncFailed != null && isSyncFailed) {
```

Comment:
The reason I decided to move `blockchainState` to the view model is that fragments/activities don't really need to know about it. They should only care about the state of UI - synced/failed, etc.